### PR TITLE
DE29441 - Fix Dynamic Content in Shadow DOM

### PR DESCRIPTION
**Pull request by svanherk**
This is based on @omsmith's PR here (https://github.com/BrightspaceUI/more-less/pull/20), but adjusted to still support the old shady DOM method.  I also hooked up sauce and wrote some basic tests around this functionality.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,2 +1,11 @@
-language: node_js
-node_js: node
+language: node_js
+node_js:
+- '9'
+script:
+- npm run test:lint
+- polymer test --skip-plugin sauce
+- 'if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then polymer test --skip-plugin local; fi'
```
Comment (dlockhart) [inline .travis.yml]: I [read recently](http://steven.casagrande.io/articles/travis-ci-and-if-statements/) that things run inside `if`-statements when they fail don't actually impact the exit code, and therefore don't fail the build. I'm not sure if this is still true (that blog post is 2 years old), but they recommend adding `|| travis_terminate 1` to fix it.

File: d2l-more-less.html [hunk 2]
```diff
@@ -338,7 +348,20 @@
...
+			if (this.__contentSlot) {
+				var children = D2L.Dom.getComposedChildren(this.__contentSlot);
+				for (var i = 0; i < children.length; ++i) {
+					this.__observer.observe(children[i], {
+						childList: true,
+						subtree: true,
+						characterData: true,
+						attributes: true
+					});
+				}
```
Comment (dlockhart) [inline d2l-more-less.html]: This is going to observe all the children of the `<slot>`, but I'm curious if we also need to observe the slot itself? Like what if one of the children gets removed or a new child added?

Comment (dlockhart) [reply d2l-more-less.html]: Nevermind, I see we wire up a separate observer right below this. Ignore me. :)

File: .travis.yml (hunk 1 above)
Comment (svanherk) [reply .travis.yml]: Interesting, it seems to have worked here: https://travis-ci.org/BrightspaceUI/more-less/builds/373970065?utm_source=github_status&utm_medium=notification (test timing looks flaky on sauce, going to see if there's a better way I can tell the animation is over).  I wonder if it's only multiple commands in the if statement?

Comment (dlockhart) [reply .travis.yml]: Yeah totally possible it only impacts multi-command statements.

File: .travis.yml [hunk 3]
```diff
@@ -1,2 +1,11 @@
-language: node_js
-node_js: node
+language: node_js
+node_js:
+- '9'
+script:
+- npm run test:lint
+- polymer test --skip-plugin sauce
```
Comment (dlockhart) [inline .travis.yml]: Nit: these 2 lines could just be `npm test`

File: wct.conf.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+	"plugins": {
+		"local": {
+			"browsers": ["chrome"]
```
Comment (dlockhart) [inline wct.conf.json]: Worth giving headless Chrome a shot? Should just be a matter of [adding these options](https://github.com/Brightspace/d2l-my-courses-ui/blob/master/wct.conf.json#L8)

File: .travis.yml (hunk 1 above)
Comment (svanherk) [reply .travis.yml]: Doesn't really hurt to add it anyways while I'm here

File: .travis.yml (hunk 3 above)
Comment (svanherk) [reply .travis.yml]: Haha nice catch!

File: wct.conf.json (hunk 4 above)
Comment (svanherk) [reply wct.conf.json]: Yup will add, need to investigate why local chrome is failing in Travis anyways

**Pull request merged**